core/templates/base-components/base-content.component.spec.ts

```diff
@@ -61,6 +62,13 @@ describe('Base Content Component', () => {
     }
   }
 
+  class MockRouteService {
+    public events: Observable<NavigationEnd> = (
```

Comment:
This looks quite complex. Can you add some comments explaining it in detail so that new readers know what's going on?

Reply:
Done.